addons/website_sale/models/product_template.py

```diff
@@ -300,7 +300,11 @@ def _get_sales_prices(self, website):
                         base_price, currency, product_taxes, taxes, self, website=website,
                     )
 
-            if not base_price and comparison_prices_enabled and template.compare_list_price:
+            if (
+                comparison_prices_enabled
+                and template.compare_list_price
+                and self.env.user.has_group('website_sale.group_product_price_comparison')
```

Comment:
Means we should not display comparison price to portal user or user without sign-up?
seems odd and not right :worried:

Reply:
I mistakenly added the group here again, its already checked in the condition through `comparison_prices_enabled`.  

Basically this behavior was intentional earlier as in previous task it was decided that - pricelist takes precedence over the comparison price and if pricelist is increasing the price we don't show strikethrough price. 

But now it is decided that we can show that on the product page irrespective of the priority, hence we don't need the condition to check that